app/sequencescape/sequencescape/api/v2/shared/has_workline_identifier.rb

```diff
@@ -11,12 +11,28 @@ module HasWorklineIdentifier
     # Existing use at time of writing is in the top right part of plate labels.
     # In future it could also be included on tube labels if needed.
     # Uses alternative_workline_identifier from the purpose config if present, otherwise uses the stock plate.
+    # NB. the config value can be an array, in the case where there is more than one possible purpose ancestor.
+    # In this case, the first one found is used.
+    #
+    # rubocop:disable Metrics/MethodLength
     def workline_reference
       alternative_workline_identifier_purpose = SearchHelper.alternative_workline_reference_name(self)
       return stock_plate if alternative_workline_identifier_purpose.nil?
 
-      ancestors.where(purpose_name: alternative_workline_identifier_purpose).last
+      if alternative_workline_identifier_purpose.is_a?(Array)
+        # Try each purpose name in the array in order until we find a match
+        alternative_workline_identifier_purpose.each do |purpose_name|
+          reference = ancestors.where(purpose_name:).last
+          return reference if reference.present?
+        end
+        # If no matches found in the array, fall back to stock plate
+        stock_plate
+      else
+        # Original behavior for a single purpose name
+        ancestors.where(purpose_name: alternative_workline_identifier_purpose).last
+      end
```

Comment:
Just wondering if this file actually needed any code changes -

`ancestors.where(purpose_name: alternative_workline_identifier_purpose).last`

'where' works with arrays as well as single items, doesn't it?

I expect something blew up though, since you're changing it!